src/cargo/ops/vendor.rs

```diff
@@ -392,6 +398,114 @@ fn cp_sources(
     Ok(())
 }
 
+fn prepare_for_vendor(
```

Comment:
Even if we plan to revisit and refactor, a notice here could be useful for people about to diverge the logic to think twice.

```suggestion
/// This is a hack for resolving cargo/#14348.
/// It basically copies the normalization logic from `prepare_for_publish`.
/// TODO: need to revisit and consolidate both for `cargo package` and `cargo vendor`. 
fn prepare_for_vendor(
```

Reply:
`r=weihanglo` whenever you feel ready. The comment is not necessary.